src/Mod/Sketcher/Gui/DrawSketchHandlerArcOfHyperbola.h

```diff
@@ -102,8 +102,9 @@ class DrawSketchHandlerArcOfHyperbola: public DrawSketchHandler
             double angleatpoint = acosh(((onSketchPos.x - centerPoint.x) * cos(phi)
                                          + (onSketchPos.y - centerPoint.y) * sin(phi))
                                         / a);
-            double b = (onSketchPos.y - centerPoint.y - a * cosh(angleatpoint) * sin(phi))
-                / (sinh(angleatpoint) * cos(phi));
+            double b = ((onSketchPos.y - centerPoint.y) * cos(phi)
+                        - (onSketchPos.x - centerPoint.x) * sin(phi))
+                / sinh(angleatpoint);

```

Comment:
sinh can also be zero. any reason why this is guaranteed to not happen?

Reply:
_sinh(theta)_ is zero if the cursor position momentarily lies exactly on the centerPoint of the hyperbolic arc.  Then _b =0/0 =NaN_
The lines of code that follow I believe draw the arc and explicitly avoid drawing one of zero length by requiring _b != NaN_, so I think the existing code already deals with this possibility.

Reply:
It seems that `angleatpoint` is zero only when the point is on `axisPoint`, not `centerPoint`, though that still gives nan. However, the check was still there even with the `cos(phi)` in the denominator.

That said there is probably some inherent property avoiding such cases.

Reply:
OK. The inherent property is simply that the hyperbola has to be on the "other side" of the second point. If it does, a different check is triggered.

Reply:
Agreed.  The code cleverly enforces that the hyperbola has to be on the "other side" of the second point by requiring _b != NaN_ . In the old code this occurred three ways:

1.  _angleatpoint = NaN_   (The acosh fails)
2. _angleatpoint = 0_         (Divide by zero)
3. _cos(phi) =0 _ (Divide by zero)

(1) and (2) create the desired behavior. (3) creates the reported bug.  The new, algebraically equivalent expression for b fixes this.

By the way, I have since managed to compile on my Mac by learning how to make an Ubuntu virtual machine and compiling there.  The patch does fix the bug, as I anticipated.